android/library/src/main/java/com/liferay/mobile/screens/ddl/model/DateField.java

```diff
@@ -66,29 +65,17 @@ protected Date convertFromString(String stringValue) {
             return null;
         }
 
-        try {
-            int lastSeparator = stringValue.lastIndexOf('/');
-
-            if (stringValue.contains("-")) {
-                SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd", getCurrentLocale());
-                simpleDateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
-                return simpleDateFormat.parse(stringValue);
-            } else if (lastSeparator == -1) {
-                return new Date(Long.parseLong(stringValue));
-            } else if (stringValue.length() - lastSeparator - 1 == 2) {
-                SimpleDateFormat simpleDateFormat = new SimpleDateFormat("MM/dd/yy", getCurrentLocale());
-                simpleDateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
-                return simpleDateFormat.parse(stringValue);
-            } else {
-                SimpleDateFormat simpleDateFormat = new SimpleDateFormat("MM/dd/yyyy", getCurrentLocale());
-                simpleDateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
-
-                return simpleDateFormat.parse(stringValue);
-            }
-        } catch (ParseException e) {
-            LiferayLogger.e("Error parsing date " + stringValue);
+        int lastSeparator = stringValue.lastIndexOf('/');
+
+        if (stringValue.contains("-")) {
+            return convertDate("yyyy-MM-dd", stringValue);
+        } else if (lastSeparator == -1) {
+            return new Date(Long.parseLong(stringValue));
+        } else if (stringValue.length() - lastSeparator - 1 == 2) {
```

Comment:
Can you extract this to a function like `isSimpleYearFormat`  or something like that so we have some semantics here? :)